components/common-go/log/fields.go

```diff
@@ -0,0 +1,118 @@
+// Copyright (c) 2023 Gitpod GmbH. All rights reserved.
+// Licensed under the GNU Affero General Public License (AGPL).
+// See License.AGPL.txt in the project root for license information.
+
+package log
+
+import log "github.com/sirupsen/logrus"
+
+const (
+	UserIDField = "userId"
+	// OwnerIDField is the log field name of a workspace owner
+	OwnerIDField = UserIDField
+	// WorkspaceIDField is the log field name of a workspace ID (not instance ID)
+	WorkspaceIDField = "workspaceId"
+	// WorkspaceInstanceIDField is the log field name of a workspace instance ID
+	WorkspaceInstanceIDField = "instanceId"
+	// ProjectIDField is the log field name of the project
+	ProjectIDField = "projectId"
+	// TeamIDField is the log field name of the team
+	TeamIDField = "teamId"
+
+	OrganizationIDField = "orgId"
+
+	ServiceContextField        = "serviceContext"
+	PersonalAccessTokenIDField = "patId"
+	OIDCClientConfigIDField    = "oidcClientConfigId"
+)
+
+// OWI builds a structure meant for logrus which contains the owner, workspace and instance.
+// Beware that this refers to the terminology outside of wsman which maps like:
+//
+//	owner = owner, workspace = metaID, instance = workspaceID
+func OWI(owner, workspace, instance string) log.Fields {
+	return log.Fields{
+		OwnerIDField:             owner,
+		WorkspaceIDField:         workspace,
+		WorkspaceInstanceIDField: instance,
+	}
+}
+
+// LogContext builds a structure meant for logrus which contains the owner, workspace and instance.
+// Beware that this refers to the terminology outside of wsman which maps like:
+//
+//	owner = owner, workspace = metaID, instance = workspaceID
+func LogContext(ownerID, workspaceID, instanceID, projectID, orgID string) log.Fields {
+	return Compose(
+		WorkspaceOwner(ownerID),
+		WorkspaceID(workspaceID),
+		WorkspaceInstanceID(instanceID),
+		ProjectID(projectID),
+		OrganizationID(orgID),
+	)
+}
+
+func WorkspaceOwner(owner string) log.Fields {
+	return String(OwnerIDField, owner)
+}
+
+func WorkspaceID(workspaceID string) log.Fields {
+	return String(WorkspaceIDField, workspaceID)
+}
+
+func WorkspaceInstanceID(instanceID string) log.Fields {
+	return String(WorkspaceInstanceIDField, instanceID)
+}
+
+func ProjectID(projectID string) log.Fields {
```

Comment:
These helpers allow us to control centrally the common fields, and be able to adjust the naming of the fields should we choose to do so.